### PR TITLE
refactor(examples): change file-sharing from async-std to tokio

### LINKED ISSUE #5180: Refactor file-sharing example to use tokio instead of async-std
### Description

The file-sharing example currently uses `async-std`, but would prefer to use `tokio`. I'd be happy to open a PR with this change

### Motivation

This issue: https://github.com/libp2p/rust-libp2p/issues/4449

### Current Implementation

`async-std` is used instead of `tokio`

### Are you planning to do it yourself in a pull request ?

Yes

### PR DESCRIPTION
**Pull request by bklaing2**
## Description

Closes: https://github.com/libp2p/rust-libp2p/issues/5180
Related: https://github.com/libp2p/rust-libp2p/issues/4449.

## Change checklist

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] A changelog entry has been made in the appropriate crates

### CODE REVIEW

File: examples/file-sharing/src/network.rs [hunk 1]
```diff
@@ -1,5 +1,5 @@
-use futures::channel::{mpsc, oneshot};
-use futures::prelude::*;
+use tokio::sync::{mpsc, oneshot};
```
Comment (dariusc93) [inline examples/file-sharing/src/network.rs]: Not sure if it matters to others, but `futures` channels can still be used here.

File: examples/file-sharing/src/network.rs [hunk 2]
```diff
@@ -67,8 +67,8 @@ pub(crate) async fn new(
         .kademlia
         .set_mode(Some(kad::Mode::Server));
 
-    let (command_sender, command_receiver) = mpsc::channel(0);
-    let (event_sender, event_receiver) = mpsc::channel(0);
+    let (command_sender, command_receiver) = mpsc::channel(100);
+    let (event_sender, event_receiver) = mpsc::channel(100);
```
Comment (dariusc93) [inline examples/file-sharing/src/network.rs]: Leaving the buffer set to 0 allows backpressure (though I dont think tokio channels allow a 0 buffer while futures channel does)

**Review by dariusc93 — COMMENTED** [review]
Left a couple of comments but overall LGTM

File: examples/file-sharing/src/network.rs (hunk 1 above)
Comment (bklaing2) [reply examples/file-sharing/src/network.rs]: I can switch it back to `futures` if preferred

File: examples/file-sharing/src/network.rs (hunk 2 above)
Comment (bklaing2) [reply examples/file-sharing/src/network.rs]: Yes, the tokio channels didn't work with a buffer size of 0

**Review by jxs — DISMISSED** [review]
LGTM

**Review by dariusc93 — APPROVED** [review]

**Review by jxs — APPROVED** [review]

**Comment by jxs** [comment]
Thanks! :heart:  
Btw @bklaing2  would you be up for updating the original #4449 issue with the tracklist of examples and tutorials already with 
`tokio` and the remaining ones still with `async-std`?

**Pull request merged**